### ISSUE TITLE
Add a custom scorer for the LIFT metric

### ISSUE THREAD

**Issue by rasbt** [rasbt/mlxtend#65]
- build a scikit-learn compatible custom scorer for computing the [LIFT](https://en.wikipedia.org/wiki/Lift_%28data_mining%29) metric

**Comment by Searil** [comment]
Hi Sebastian,

I heavily use scikit-learn in my works and recently I found mlxtend. 

I want to work on this issue and implement this feature. 

Could you please make a brief explanation about the issue details?

**Comment by rasbt** [comment]
Hi, Batuhan,
thanks for your interest in helping with the implementation. What I had in mind was a `lift_score` analogous to scikit-learn`s `accuracy_score` (and other scorers) from the sklearn.metrics module. However, to make the `lift_score` really useful, it should be compatible with GridSearchCV as well.

For instance, it should support the following (think of `lift_score` instead of `accuracy_score`)

```python
from sklearn import svm, datasets
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import accuracy_score
from sklearn.metrics import make_scorer

acc_scorer = make_scorer(accuracy_score)
iris = datasets.load_iris()
parameters = {'kernel':('linear', 'rbf'), 'C':[1, 10]}

svc = svm.SVC()
clf = GridSearchCV(svr, parameters, scoring=acc_scorer)
clf.fit(iris.data, iris.target)
```

and the standalone:

```python
y_pred = svc.fit(X, y).predict(X)
accuracy_score(y_pred, y)
```

I was thinking to add this to https://github.com/rasbt/mlxtend/tree/master/mlxtend/evaluate either as `lift_score` function in `lift_score.py` or `scoring.py`.

**Comment by Searil** [comment]
Okay Sebastian, I understand what you think and it makes sense for me. I know what lift metric is but to be sure from everything now I am reading some articles about it. I will plan to add this function to `scoring.py`.

Also could you please assign me to this issue?

**Comment by rasbt** [comment]
Cool! I think LIFT is not as common in academia, but it is often used in industrial applications -- there were some people who asked me about it at the scikit workshops at leaset. Hm, I think I cannot assign you to this issue via the GitHub API (I think it only works for repository admins), but I set the label to "in progress" instead.

**Comment by Searil** [comment]
Hi Sebastian,

Based on this [article](http://www.kdnuggets.com/2016/04/association-rules-apriori-algorithm-tutorial.html) and the book(Data Science and Big Data Analytics - Chapter 5) that I read, I implemented a simple lift metric which you can find it in my repo via this [link](https://github.com/Searil/lift_metric/blob/master/Lift%20Metric.ipynb).

Could you please give me some feedback when you look at it ? (As a reminder, I coded this notebook only for showing lift metric works correct or not for simple dataset. When we agree on everything, I will make it compatible with mlxtend. )

**Comment by rasbt** [comment]
Closing. Has been addressed in #136 :)
